### ISSUE TITLE
Remove UA sniff in sce.js

### ISSUE THREAD

**Issue by david-mark** [angular/angular.js#15613]
***Note*: for support questions, please use one of these channels: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#question. This repository's issues are reserved for feature requests and bug reports.**

**Do you want to request a *feature* or report a *bug*?**

Neither. UA sniffing and either comments or quirks mode test is invalid.

```javascript
    // Support: IE 9-11 only
    // Prereq: Ensure that we're not running in IE<11 quirks mode.  In that mode, IE < 11 allow
    // the "expression(javascript expression)" syntax which is insecure.
    if (enabled && msie < 8) {
      throw $sceMinErr('iequirks',
        'Strict Contextual Escaping does not support Internet Explorer version < 11 in quirks ' +
        'mode.  You can fix this by adding the text <!doctype html> to the top of your HTML ' +
        'document.  See http://docs.angularjs.org/api/ng.$sce for more information.');
    }
```

Testing for an IE version < 8 is not the correct test for quirks mode. Assuming that `msie` is based on the `documentMode` property, the correct check would be:

```javascript
if (enabled && msie == 5) {
```

**Comment by gkalpak** [comment]
[Since IE10][1], there is "Quirks" mode and there is "IE5 Quirks mode". But I think this comment is just inaccurately using "Quirks mode" to refer to IE <8, because the mentioned vulnerability (CSS expressions) is present in versions [up to IE7][2].

So, the code works as expected, but the comment and the [corresponding error page][3] could be improved.
PRs welcome :smiley:

[1]: https://blogs.msdn.microsoft.com/ie/2011/12/14/interoperable-html5-quirks-mode-in-ie10/
[2]: https://blogs.msdn.microsoft.com/ie/2008/10/16/ending-expressions/
[3]: https://github.com/angular/angular.js/blob/5b6763f575b0e824b5c566351e0135898264ee7b/docs/content/error/%24sce/iequirks.ngdoc

**Comment by david-mark** [comment]
Right, MS is using "quirks mode" incorrectly to mean IE < 8. Never noticed that before.

As for Angular, adding a doctype isn't the correct advice either. Suppose the correct advice is to get out of IE 7 or either of the quirks mode compatibility views. Assume neither of those are supported.

Will issue a PR.

**Comment by gkalpak** [comment]
Latest Angular versions do not support IE <9, so this shouldn't be a big concern. This is either a leftover form a very, very old version that did support IE <8 or just an attempt to be extra cautious on security-related matters.

Looking forward to the PR :smiley:
